### PR TITLE
Javascript Tic Tac toe game

### PR DESCRIPTION
**Pull request by gitnyasha**
- created a tic tac toe game with html, css and javascript
- we placed our code into modules in the javascript file
- in the game you can enter your names, player 1 is x and player 2 is o
- the goal is to match 3 straight symbols to win

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      "after",
+      {
+        "overrides": {
+          ":": "before"
+        }
+      }
+    ],
+    "object-shorthand": ["always"]
+  }
+}
```
Comment (bolah2009) [inline .eslintrc.json]: Is there a reason why this rules are changed? 🤔 

@bellom and @fatymahmed requested to change this in https://github.com/gitnyasha/tic-tac-toe-javascript/pull/1#discussion_r428585566

Please let's use the [**correct config files**](https://github.com/microverseinc/linters-config/tree/master/javascript).

> Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))

File: README.md [hunk 2]
```diff
@@ -1,8 +1,6 @@
 # Library
```
Comment (bolah2009) [inline README.md]: Can we change this? I am sure this is not a `Library` app... 🤔

File: README.md [hunk 3]
```diff
@@ -33,7 +31,7 @@ Download the project
```
Comment (bolah2009) [inline README.md:46]: Please change this instruction:
```md
- That's it. Start saving your books.
```

File: README.md [hunk 4]
```diff
@@ -33,7 +31,7 @@ Download the project
 
 👤 \*\*\*\*
 
-- Github: [](https://github.com/)
+- Github: [konyan](https://github.com/konyan)
 - Twitter: [](https://twitter.com/)
 - Linkedin: [](https://www.linkedin.com/)
```
Comment (bolah2009) [inline README.md]: [Optional]

Please, let's update these links to an actual profile link. Also, like `**Marshall Nyasha Chikari**` was added below add your name too to the readme @konyan.

File: app.js [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+              board[6].innerHTML === board[7].innerHTML &&
+              board[7].innerHTML === board[8].innerHTML &&
+              board[6].innerHTML.trim() !== ""
+            ) {
+              showWinner(game);
+            } else if (
+              board[0].innerHTML === board[3].innerHTML &&
+              board[3].innerHTML === board[6].innerHTML &&
+              board[0].innerHTML.trim() !== ""
+            ) {
```
Comment (bolah2009) [inline app.js]: I think it is important not to miss up the app logic with DOM.

We can create an array (or object) in the `gameBoard` function that stores the plays from each player and a method that keeps tracks of this array to know when there's a winner or a draw.

This information should be enough to be used to update the state of the app whenever a player makes a **valid move**, we can also add feedback for **invalid moves**.


> * [ ] It has a function with the only responsibility of rendering the contents of the gameboard array to the webpage ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/render_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript))
> * [ ] It has a function that that allows a player to 'mark' a spot on the board ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/mark_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript))
> * [ ] It has a method(s) to check the game's winning conditions ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/win_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript))


This was also discussed by previous reviewers in https://github.com/gitnyasha/tic-tac-toe-javascript/pull/1#discussion_r428587246

File: app.js [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+  return {
+    showBoard: function (game) {
+      boardWinner(game);
+    },
+
+    newGame: function (xPlayer, oPlayer) {
+      return new GameBoard(xPlayer, oPlayer);
+    },
+  };
+})();
```
Comment (bolah2009) [inline app.js]: [Optional]

Some feedback that should be added/removed
- When there's a winner, let's remove the information that the next player should make their move (`Now your turn !`)
   image
- This is the same for `draw`, let's hide this information and announce that the game is a draw
   image

**Review by bolah2009 — CHANGES_REQUESTED** [review]
Good job with the game...

Please let's check the comments below and makes changes accordingly.


Kindly note that closing PR (#1) may affect consistency in reviews if the current reviewer did not know about what was suggested in previous reviews. Please, let's use the same PR for all review cycles instead of closing/merging PRs before the project is approved.


#### :recycle:  Make these changes and submit another code review request. 👍🏾

#### Feel free to reach me on slack if you having issues implementing these features. :relaxed: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Review by fatymahmed — CHANGES_REQUESTED** [review]
## Hi @gitnyasha @konyan, 
Good job making some of the requested changes 👏, but for this project to be approved, please make all the requested changes.
 
See my comments below.

### Status: Changes Required ♻️

File: .eslintrc.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    "eol-last": "off",
+    "arrow-parens": "off",
+    "max-classes-per-file": ["error", 2],
+    "quotes": [
+      2,
+      "double",
+      {
+        "avoidEscape": true
+      }
+    ]
```
Comment (fatymahmed) [inline .eslintrc.json]: As mentioned by @bolah2009, please do not change the linter config files, it's also mentioned on Pathwright, see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png), use [these](https://github.com/microverseinc/linters-config/tree/master/javascript) linter config files that are provided by microverse.

Comment (gitnyasha) [reply .eslintrc.json]: Whenever we save our code it places double quotes automatically which lint is picking as errors so I don’t know what you want us to do

File: app.js [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+  const showWinner = (game) => {
+    document.getElementById("winner").innerHTML = game.currentPlay.name === game.oPlayer.name
+      ? game.xPlayer.name
+      : game.oPlayer.name;
+
+    document.getElementById("message").style.display = "block";
+    document.getElementById("game-status").innerHTML = "status = game over";
+    game.status = "Game Over";
+    document.getElementById("reset").classList = "";
+  };
```
Comment (fatymahmed) [inline app.js]: [OPTIONAL]
As mentioned by @bolah2009, you could remove the "Now you turn !" text when the game ends, that is when a player wins or when there are no more possible moves to make and there's no winner (a tie)

Comment (gitnyasha) [reply app.js]: Its fixed now

File: .eslintrc.json (hunk 7 above)
Comment (fatymahmed) [reply .eslintrc.json]: okay, now I get why you kept changing the rules 🙂, one way to solve this is to use `.prettierrc.js` file with the following content `module.exports = {
  singleQuote: true,
};
` but maybe there's a better way to fix this, let's see what @bolah2009 says about this

Comment (bolah2009) [reply .eslintrc.json]: I agree with @fatymahmed here. If prettier is your formatting tool, I think that can solve the issue you having.

Comment (gitnyasha) [reply .eslintrc.json]: we solved the issue

**Review by Torres-ssf — APPROVED** [review]
### Project Approved :+1: :+1: :+1: 
Hi team :hand: 
The required changes were implemented.
Good luck with the next project.

Happy coding :computer: 
**Sergio Torres**

**Pull request merged**